packages/apollo-voyager-conflicts/src/states/HashObjectState.ts

```diff
@@ -1,27 +1,54 @@
-import * as debug from 'debug'
+import { ConflictLogger } from '../api/ConflictLogger'
 import { ObjectState } from '../api/ObjectState'
 import { ObjectStateData } from '../api/ObjectStateData'
-import { CONFLICT_LOGGER } from '../constants'
+import { StatePersistence } from '../api/StatePersistence'
 
 /**
  * Object state manager using a hashing method provided by user
  */
 export class HashObjectState implements ObjectState {
+  private logger: ConflictLogger | undefined
   private hash: (object: any) => string
+  private statePersistence?: StatePersistence
 
   constructor(hashImpl: (object: any) => string) {
     this.hash = hashImpl
   }
 
   public hasConflict(serverData: ObjectStateData, clientData: ObjectStateData) {
     if (this.hash(serverData) !== this.hash(clientData)) {
+      if (this.logger) {
+        this.logger.info(`Conflict when saving data.
+        current: ${serverData}, client: ${clientData}`)
+      }
```

Comment:
is the `clientData` here is the one that before changes are made (currentState -1 ), or after the changes are made (currentState)? If it is actually the current state, doesn't it mean this will always cause a conflict?

Reply:
or it depends on how the `hash` function is implemented: it needs to return a value that matches the current server data? Do you have an example of such a hash function?

Reply:
Framework gives many options for integration. Currently we using VersionedObjectState (see VE aproach and we used it in example application in this repo and also on ionic example.
However integration tests are going to give us examples for entire api that is provided.

I identified some flaws in API here and going to get this fixed. 
Also working on diagram for this idea